leetcode_python/Greedy/maximum-number-of-non-overlapping-palindrome-substrings.py

```
@@ -0,0 +1,75 @@
+#
+#   then one pass over the ends applies the greedy: if best_start[e] sits
+#   strictly after the last taken end, take it.
+#
+# time = O(n^2), space = O(n)
+class Solution(object):
+    def maxPalindromes(self, s, k):
+        n = len(s)
+        best_start = [-1] * n          # largest l with s[l..e] a palindrome, len >= k
+
+        for center in range(2 * n - 1):
+            l = center // 2
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)